infra/main.tf

```diff
@@ -0,0 +1,190 @@
+terraform {
+  required_providers {
+    google = {
+      source = "hashicorp/google"
+      version = "3.48.0"
+    }
+    kubernetes = {
+      source = "hashicorp/kubernetes"
+      version = "1.13.3"
+    }
+  }
+}
+
+variable "gcp_project" {}
+variable "gcp_region" {}
+variable "gcp_zone" {}
+variable "domain" {}
+
+provider "google" {
+  credentials = file("~/.hail/terraform_sa_key.json")
+
+  project = var.gcp_project
+  region  = var.gcp_region
+  zone    = var.gcp_zone
+}
+
+data "google_client_config" "provider" {}
+
+resource "google_project_service" "service_networking" {
+  service = "servicenetworking.googleapis.com"
+}
+
+resource "google_compute_network" "internal" {
+  name = "internal"
+}
+
+data "google_compute_subnetwork" "internal_default_region" {
+  name = "internal"
+  region = var.gcp_region
+  depends_on = [google_compute_network.internal]
+}
+
+resource "google_container_cluster" "vdc" {
+  name = "vdc"
+  location = var.gcp_zone
+  network = google_compute_network.internal.name
+
+  # We can't create a cluster with no node pool defined, but we want to only use
+  # separately managed node pools. So we create the smallest possible default
+  # node pool and immediately delete it.
+  remove_default_node_pool = true
+  initial_node_count       = 1
+
+  master_auth {
+    username = ""
+    password = ""
+
+    client_certificate_config {
+      issue_client_certificate = false
+    }
+  }
+}
+
+resource "google_container_node_pool" "vdc_preemptible_pool" {
+  name       = "preemptible-pool"
+  location   = var.gcp_zone
+  cluster    = google_container_cluster.vdc.name
+
+  autoscaling {
+    min_node_count = 0
+    max_node_count = 200
+  }
+
+  node_config {
+    preemptible  = true
+    machine_type = "n1-standard-2"
+
+    metadata = {
+      disable-legacy-endpoints = "true"
+    }
+
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/cloud-platform"
+    ]
+  }
+}
+
+resource "google_container_node_pool" "vdc_nonpreemptible_pool" {
+  name       = "nonpreemptible-pool"
+  location   = var.gcp_zone
+  cluster    = google_container_cluster.vdc.name
+
+  autoscaling {
+    min_node_count = 0
+    max_node_count = 200
+  }
+
+  node_config {
+    preemptible  = false
+    machine_type = "n1-standard-2"
+
+    metadata = {
+      disable-legacy-endpoints = "true"
+    }
+
+    oauth_scopes = [
+      "https://www.googleapis.com/auth/cloud-platform"
+    ]
+  }
+}
+
+resource "google_compute_global_address" "db_ip_address" {
```

Comment:
Should this be using `google_compute_address`?

Reply:
So I couldn't get that to work.  The documentation wasn't super explicit about this, but as far as I can tell, the Could SQL database wants to be on the global rather than regional network.